src/System.Text.Json/src/System/Text/Json/Document/JsonDocument.Parse.cs

```diff
@@ -353,11 +376,14 @@ private static async
 
             try
             {
+                // Save the length to a local to be reused across awaits.
+                int utf8BomLength = JsonConstants.Utf8Bom.Length;
```

Comment:
Why is this necessary? Is it costly to access the span/length each time?

Reply:
The alternative to this is to make each place that uses it read the property and access the length field.  While the JIT could do magic there, sharplab said that accessing JsonConstants.Utf8Bom.Length each time made the async state machine's MoveNext be larger.

Reply:
Really? I would have thought the other way around since the local now needs to survive across the async/await boundary. cc @stephentoub

Reply:
[sharplab.io](https://sharplab.io/#v2:EYLgHgbALANALiAhgZwLYB8ACAGABJgRgG4BYAKEwCZtzMBmfAifSxgdlwG9zdf8HCzTAA5GAVgB0mZgFlEASwB2ACgCUXHny2YAnLhkBTOAAsA9gBMCyxQYDuuYAE84BgNoFs2ALoxcN+4QSAJIA8hKGqKYATo4AynBRBoioaqqkZFoAvpq8ObgADlHyAG6ILoxCooSS0gA8SnAAfPpGZpbKTi6uXriJii7mvoGhEvGJybjICUmoqnncGVp8DbgArnAAZsIAQqaoADIGigDmJrgAvLh06UvL/bi2RXAuihe42De3uCsANihwACUkuY3h88lpzKZwXwFl9bn8pkDECDLrpJtNkhIkeYAILIRyKADGymhcN4fQGMFJZMe8meRypizJS3WW12ByOp2MuAAtA8ni9VBIAMKmRQbeTHVaJHG2BRwZQbRA/ZAGNLkalLWn014AakuCMBwM+t0yD2M8h+BlwykN2NwzTwADInfy6S9cLU1psdntDicTOqmUtqZgOHbjXlssG8oUSmVrYJ8FUCDUIPV+s1DCYLJQOs43D0KQZBoxgmExjN0eNZvNqSttR7LmDg1pfv97c2TUtIdTYczeBHkW80VMa1jgXiCcTNV9i4NZ7dGwzF0sGHzl4ohaLxZLpQZZfLFcrVUHV26dbh9bgh+Zu1lzZbrbaO8CHe9cC6Lx6vXQg3DQ3DV9kW7aNMiAA==)

Before - 32 bytes:
```text
Program+<Method1>d__1.SetStateMachine(System.Runtime.CompilerServices.IAsyncStateMachine)
    L0000: sub rsp, 0x28
    L0004: add rcx, 0x20
```
After - 24 bytes:
```text
Program+<Method2>d__2.SetStateMachine(System.Runtime.CompilerServices.IAsyncStateMachine)
    L0000: sub rsp, 0x28
    L0004: add rcx, 0x18
...
```
+ 4 fewer instructions in MoveNext (if the length was just a regular integer literal).
Accessing the span property + length does increase the instruction count.